telos-frontend/src/components/todo/DashboardTodo.jsx

```diff
@@ -105,6 +143,48 @@ const DashboardTodo = () => {
     alert('Please enter something :)');
   };
 
+  const cancelevent = () => {
+    const currentIndex = cancel.indexOf(reItem.name);
+    const newCancel = [...cancel];
+
+    if (currentIndex === -1) {
+      newCancel.push(reItem.name);
+    } else {
+      newCancel.splice(currentIndex, 1);
+    }
+    setCancel(newCancel);
+    setAnchorEl(null);
+  };
+
+  const deleteevent = () => {
+    const newList = [...newItem];
+    for (const x of newList) {
+      if (x.name === reItem.name) {
+        const index = newList.indexOf(x);
+        newList.splice(index, 1);
+      }
+    }
+
+    setNewItem(newList);
+    setItemList(newList);
+    setAnchorEl(null);
+  };
+
+  const scheduleevent = () => {
+    const newList = [...newItem];
+    const reItemx = reItem;
```

Comment:
I feel that this probably isn't the best way to name the variable. Also, this probably isn't a deep copy ?